File: packages/typia/src/transformers/features/http/HttpAssertQueryTransformer.ts
import { HttpAssertQueryProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace HttpAssertQueryTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "http.assertQuery",
      write: HttpAssertQueryProgrammer.write,
    });
}

File: packages/transform/src/features/http/HttpAssertQueryTransformer.go
package http

import (
	corehttp "github.com/samchon/typia/packages/core/src/programmers/http"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var HttpAssertQueryTransformer = httpAssertQueryTransformerNamespace{}

type httpAssertQueryTransformerNamespace struct{}

func (httpAssertQueryTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return corehttp.HttpAssertQueryProgrammer.Write(call.TypeText)
}
